### ISSUE TITLE
Why do we need padding=100 for a filter of size 3?

### ISSUE THREAD

**Issue by deartonym** [wkentaro/pytorch-fcn#40]
As the title, in torchfcn/models/fcn32s.py we have the setting for the first conv1 layer:

`nn.Conv2d(3, 64, 3, padding=100),`

Why do we need a padding of side length 100 instead of 1 according to the filter size 3?

Thanks

**Comment by Viresh-R** [comment]
Shelhamer and Long use a padding of 100 for the first conv layer in their Caffe implementation.
https://github.com/shelhamer/fcn.berkeleyvision.org/blob/master/pascalcontext-fcn8s/train.prototxt

**Comment by wkentaro** [comment]
Thank you @Viresh-DL

**Comment by weiaicunzai** [comment]
if anyone interested:

Why pad the input?: The 100 pixel input padding guarantees that the network output can be aligned to the input for any input size in the given datasets, for instance PASCAL VOC. The alignment is handled automatically by net specification and the crop layer. It is possible, though less convenient, to calculate the exact offsets necessary and do away with this amount of padding.

https://github.com/shelhamer/fcn.berkeleyvision.org

**Comment by Niloy-Chakraborty** [comment]
I have another issue. I am writing the code in keras/ tf.keras. 
How can I add the padding=100, as it seems, keras has padding= same, valid

**Comment by WMF1997** [comment]
> I have another issue. I am writing the code in keras/ tf.keras.
> How can I add the padding=100, as it seems, keras has padding= same, valid

what about using ```tf.layers.conv2d```??
